Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -106,6 +106,10 @@ class RewriteGlobalVariables : public AST::Visitor {
     AST::StructureMember& createArgumentBufferEntry(unsigned binding, AST::Variable&);
     AST::StructureMember& createArgumentBufferEntry(unsigned binding, const SourceSpan&, const String& name, AST::Expression& type);
     void finalizeArgumentBufferStruct(unsigned group, Vector<std::pair<unsigned, AST::StructureMember*>>&);
+    void insertDynamicOffsetsBufferIfNeeded(const AST::Function&);
```

Comment:
@tadeuzagallo I had to make a few more changes because `__DynamicOffsets` was not always generated (for example, when there were no argument buffers in the fragment shader).

Change is probably substantial enough to not simply re-land, so I will wait for your input